Add new page for saving/reusing actions for apps (#31057)
DataDog/documentation: content/en/actions/app_builder/saved_actions.md

---
title: "Save and Reuse Actions"
description: Save and reuse an action and its parameters
disable_toc: false
disable_sidebar: false
algolia:
  tags: ['app builder', 'actions']
type: documentation
aliases:
  - /app_builder/actions_catalog/action_blueprints/
  - /service_management/app_builder/actions_catalog/saved_actions/
  - /service_management/app_builder/actions/saved_actions/

>   - /service_management/app_builder/saved_actions

Reviewer: Did these URLs exist before? If not, and if they're just copied from the Workflows version of this page, we probably don't need them because people won't be going to them 🙂

further_reading:
- link: "/actions/workflows/actions/"
  tag: "Documentation"
  text: "Learn about actions in workflows"
- link: "/integrations/"
  tag: "Documentation"
  text: "Learn about integrations"

---

Use the _Saved Actions_ feature to store and reuse an action and its parameters. You can insert a saved action into your app as a new step, or use one to populate an existing step's parameters.

## Save an action

> 1. From [your apps list][1], hover over an app and click the **Edit** {{< img src="icons/pencil.png" inline="true" style="width:14px;">}} icon.

Reviewer: ```suggestion
1. In Datadog, navigate to [**Actions** > **App Builder**][1], hover over an app, and click the **Edit** {{< img src="icons/pencil.png" inline="true" style="width:14px;">}} icon.
```

Reviewer: Might be worth mentioning something about permissions? I didn't have permission to edit the first many I rolled over!

> 1. Click the **Actions** {{< img src="icons/kebab.png" inline="true" style="width:14px;">}} menu, then **Save Action**.

Reviewer: Can you give some more context on where this button is? It can be easy to get lost in all the panes on these pages!

I also was confused about why you would go in to edit something, then save it without making changes. Is it like duplicating an existing action?

1. Enter a name and description for the action.

> 1. If you want anyone in your organization to have access to the action, toggle **Usable by others in the organization**.

Reviewer: ```suggestion
1. If you want others in your organization to have access to the action, turn on the **Usable by others in the organization** toggle.
```

1. Verify the configuration details for the action and click **Save Action Configuration**.

## Use a saved action in your app

> 1. In the app editor view, click the **Actions** {{< img src="icons/kebab.png" inline="true" style="width:14px;">}} menu and select **Use Saved Actions**.

Reviewer: This actions button doesn't seem to always be visible either?

> 1. In the app editor view, click the **Actions** {{< img src="icons/kebab.png" inline="true" style="width:14px;">}} menu and select **Use Saved Actions**.

Reviewer: ```suggestion
1. In Datadog, navigate to [**Actions** > **App Builder**][1], hover over an app, and click the **Edit** {{< img src="icons/pencil.png" inline="true" style="width:14px;">}} icon.
1. Click the **Actions** {{< img src="icons/kebab.png" inline="true" style="width:14px;">}} menu and select **Use Saved Actions**.
```
I'd add this step in again, so that if a user finds this section of the page using search and it anchors down directly, they still get some context about how to get there!

> 1. Browse through the list to find the Saved Action you're looking for.
> 1. Select the Saved Action to add it as a configured step in your app.

Reviewer: ```suggestion
1. Browse through the list to find the saved action you're looking for.
1. Select the saved action to add it as a configured step in your app.
```
"Saved action" only seems to be consistently capitalized when referring to UI labels and not the actual actions, so I think we might as well use sentence case otherwise?

## Manage a saved action

> You can edit, clone, or delete your Saved Actions from the Action Catalog.

Reviewer: ```suggestion
You can edit, clone, or delete your saved actions from the Action Catalog.
```

To find a saved action:

> 1. Navigate to the [**Action Catalog**][2].

Reviewer: ```suggestion
1. In Datadog, navigate to the [**Action Catalog**][2].
```
I like to always specify when we're in Datadog, especially when there could potentially be integrations involved 🙂

1. Click **Saved Actions**. 

> 1. Browse through the list of Saved Actions or use the search bar. Hover over the Saved Action you'd like to edit, clone, or delete.

Reviewer: ```suggestion
1. Browse through the list of saved actions or use the search bar to search for actions by name. Hover over the saved action you'd like to edit, clone, or delete.
```
After doing a small amount of testing, it looks like the search only looks for names? Thought it might be worth pointing out, so users aren't looking for, say, content within the actions.

1. Click **Manage Saved Actions**. 

> 1. Select the icon to edit {{< img src="icons/pencil.png" inline="true" style="width:14px;">}}, clone {{< img src="icons/clone.png" inline="true" style="width:14px;">}}, or delete {{< img src="icons/delete.png" inline="true" style="width:14px;">}} the Saved Action.

Reviewer: ```suggestion
1. Select the icon to edit {{< img src="icons/pencil.png" inline="true" style="width:14px;">}}, clone {{< img src="icons/clone.png" inline="true" style="width:14px;">}}, or delete {{< img src="icons/delete.png" inline="true" style="width:14px;">}} the saved action.
```

> **Note:** If you did not create an action, you cannot delete it or edit it directly. Instead, select the **Clone** {{< img src="icons/clone.png" inline="true" style="width:14px;">}}icon to copy the action and make your configuration changes. 

Reviewer: I would consider putting this above the procedure. That way, users wouldn't get to the end and be like "bah! Why can't I click this!" - they'd have their expectations set beforehand and know what to do before doing it 💪🏻

## Further Reading

{{< partial name="whats-next/whats-next.html" >}}

> <br>Do you have questions or feedback? Join the **#workflows** channel on the [Datadog Community Slack][3].

Reviewer: Is this line necessary on the app builder version of the page? Just wanted to check!

1. If you want anyone in your organization to have access to the action, toggle **Usable by others in the organization**.
1. Verify the configuration details for the action and click **Save Action Configuration**.

> {{< img src="service_management/workflows/save_action_1.mp4" alt="Click the Saved Action icon to save an action for later use." video="true" width=80% >}}

Author: @josephelkh good question! I had the same reasoning here.
